sdk/maps/maps-search/samples-dev/search.ts

```diff
@@ -0,0 +1,349 @@
+  StructuredAddress,
+  SearchAddressRequest,
+  ReverseSearchAddressRequest,
+  FuzzySearchRequest,
+  GeoJsonPolygonCollection,
+  GeoJsonCircleOrPolygonFeatureCollection,
+} from "@azure/maps-search";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function main() {
+  /**
+   * Azure Maps supports two ways to authenticate requests:
+   * - Shared Key authentication (subscription-key)
+   * - Azure Active Directory (Azure AD) authentication
+   *
+   * In this sample you can put MAPS_SUBSCRIPTION_KEY into .env file to use the first approach or populate
+   * the three AZURE_CLIENT_ID, AZURE_CLIENT_SECRET & AZURE_TENANT_ID variables for trying out AAD auth.
+   *
+   * More info is available at https://docs.microsoft.com/en-us/azure/azure-maps/azure-maps-authentication.
+   */
+  /** Shared Key authentication (subscription-key) */
+  const subscriptionKey = process.env.MAPS_SUBSCRIPTION_KEY || "";
+  const credential = new AzureKeyCredential(subscriptionKey);
+  const client = new MapsSearchClient(credential);
+
+  /** Azure Active Directory (Azure AD) authentication */
+  // const credential = new DefaultAzureCredential();
+  // const mapsClientId = process.env.MAPS_CLIENT_ID || "";
+  // const client = new MapsSearchClient(credential, mapsClientId);
+
+  console.log(" --- Geocode address:");
+  console.log(await client.searchAddress("400 Broad, Seattle"));
+
+  console.log(" --- Reverse-geocode coordinates to address:");
+  const coordinates: LatLon = [47.59118, -122.3327];
+  console.log(await client.reverseSearchAddress(coordinates));
+
+  console.log(" --- Reverse-geocode coordinates to cross street address:");
+  console.log(await client.reverseSearchCrossStreetAddress(coordinates));
+
+  console.log(" --- Geocode structured address:");
+  const structuredAddress: StructuredAddress = {
+    countryCode: "US",
+    streetNumber: "15127",
+    streetName: "NE 24th Street",
+    municipality: "Redmond",
+    countrySubdivision: "WA",
+    postalCode: "98052",
+  };
+  console.log(await client.searchStructuredAddress(structuredAddress));
+
+  console.log(" --- Perform a fuzzy search with coordinates:");
+  let fuzzyResult = await client.fuzzySearch({ query: "pizza", coordinates });
+  console.log(fuzzyResult);
+
+  console.log(" --- Perform a fuzzy search with country filter:");
+  fuzzyResult = await client.fuzzySearch({ query: "pizza", countryCodeFilter: ["Fr"] });
+  console.log(fuzzyResult);
+
+  console.log(" --- Perform a fuzzy search with coordinate and country filter:");
+  fuzzyResult = await client.fuzzySearch({
+    query: "pizza",
+    coordinates,
+    countryCodeFilter: ["Fr"],
+  });
+  console.log(fuzzyResult);
+
+  // let's save geometry IDs from the fuzzy search for the getSearchPolygon example
+  let geometryIds: string[] = [];
+  fuzzyResult = await client.fuzzySearch({ query: "Netherlands", countryCodeFilter: ["NL"] });
+  if (fuzzyResult.results) {
+    fuzzyResult.results.forEach((res: any) => {
```

Comment:
We don't use `SearchAddressResultItem` because we want to unify some of the interface, for example, `LatLon`
But I think we can still turn the line to `res.dataSources?.geometry?.id` with the current interface. And I think using optional chaining is better. Thanks for the advice!

Reply:
I found that the `dev-tool samples publish --force` is not support optional chaining. I am seeking for help and will come back to this later.

Reply:
oops sorry I forgot that we don't support that in samples yet.  In this case you can keep the original code. However, I still don't get why cast `rest` to `any`.  What's the issue with `fuzzyResult.results.forEach((res) => {`?

Reply:
I see what you mean. There's no need for `any` here. Let me remove it. Thanks!

Reply:
Removed in 1b1508cefaeef9abfbb010dd5e649589e3ba99b4